src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -273,7 +273,13 @@ std::shared_ptr<Realm> RealmCoordinator::get_realm(Realm::Config config, util::O
         REALM_ASSERT(!version || realm->read_transaction_version() == *version);
         return realm;
     }
-    do_get_realm(std::move(config), realm, version, lock);
+    try {
+        do_get_realm(std::move(config), realm, version, lock);
```

Comment:
do_get_realm is invoked from other methods as well. should we handle it the same?

Reply:
Good point!